spacepy/pybats/bats.py

```diff
@@ -640,40 +646,122 @@ def trace(self, bats):
 
         # Check if line is closed to body.
         if 'rbody' in bats.attrs:
-            r1 = sqrt(self.x[0]**2.0  + self.y[0]**2.0)
+            r1 = sqrt(self.x[ 0]**2.0 + self.y[0]**2.0)
             r2 = sqrt(self.x[-1]**2.0 + self.y[-1]**2.0)
             if (r1 < bats.attrs['rbody']) and (r2 < bats.attrs['rbody']):
                 self.open = False
 
     def plot(self, ax, *args, **kwargs):
         '''
-        Add streamline to axes object "ax". 
+        Add streamline to axes object "ax".
         '''
         ax.plot(self.x, self.y, self.style, *args, **kwargs)
 
+
 class Bats2d(IdlFile):
     '''
-    A child class of :class:`~pybats.IdlFile` taylored to BATS-R-US output.
+    A child class of :class:`~pybats.IdlFile` tailored to 2D BATS-R-US output.
+
+    Calculations
+    ------------
+    
+    New values can be added via the addition of new keys.  For example, 
+    a user could add radial distance to an equatorial Bats2d object as follows:
+
+    >>> import numpy as np
+    >>> from spacepy.pybats import bats
+    >>> mhd = bats.Bats2d('z=0_example.out')
+    >>> mhd['rad'] = np.sqrt( mhd['x']**2 + mhd['y']**2 )
+
+    Note, however, that if the user switches the data frame in a *.outs file
+    to access data from a different epoch, these values will need to be
+    updated.
+
+    An exception to this is built-in `calc_*` methods, which perform common
+    MHD/fluid dynamic calculations (i.e., Alfven wave speed, vorticity, and
+    more.)  These values are updated when the data frame is switched (see the
+    `switch_frame` method).
+
+    Plotting
+    --------
+
```

Comment:
Same question about whether blank lines under section headings affect the rendering in docs